storm-core/src/clj/backtype/storm/daemon/supervisor.clj

```diff
@@ -728,29 +848,92 @@
   (.shutdown supervisor)
   )
 
-(defn setup-storm-code-dir [conf storm-conf dir]
+(defn setup-storm-code-dir
+  [conf storm-conf dir]
  (if (conf SUPERVISOR-RUN-WORKER-AS-USER)
   (worker-launcher-and-wait conf (storm-conf TOPOLOGY-SUBMITTER-USER) ["code-dir" dir] :log-prefix (str "setup conf for " dir))))
 
+(defn setup-blob-permission
+  [conf storm-conf path]
+  (if (conf SUPERVISOR-RUN-WORKER-AS-USER)
+    (worker-launcher-and-wait conf (storm-conf TOPOLOGY-SUBMITTER-USER) ["blob" path] :log-prefix (str "setup blob permissions for " path))))
+
+(defn download-blobs-for-topology!
+  "Download all blobs listed in the topology configuration for a given topology."
+  [conf stormconf-path localizer tmproot]
+  (let [storm-conf (read-supervisor-storm-conf-given-path conf stormconf-path)
+        blobstore-map (storm-conf TOPOLOGY-BLOBSTORE-MAP)
+        user (storm-conf TOPOLOGY-SUBMITTER-USER)
+        topo-name (storm-conf TOPOLOGY-NAME)
+        user-dir (.getLocalUserFileCacheDir localizer user)
+        localresources (blobstore-map-to-localresources blobstore-map)]
+    (when localresources
+      (when-not (.exists user-dir)
+        (FileUtils/forceMkdir user-dir)
+        (setup-blob-permission conf storm-conf (.toString user-dir)))
```

Comment:
Minor, but I think we do not need to `setup-blob-permission` here, since we are doing it next unconditionally.